### ISSUE TITLE
Payload not updated when using request.error filter

### ISSUE THREAD

**Issue by llindemann** [directus/directus#17002]
### Describe the Bug

When updating the payload in a flow or an extension using the `request.error` filter, the changes are discarded

### To Reproduce

The following script should prevent fields with a `must be [null]` validation to display an error message to the user:

```javascript
module.exports = async function(data) {
	if (Array.isArray(data.$trigger.payload?.errors)) {
			const errors = data.$trigger.payload.errors.filter(a => !a.message.includes('" must be [null]'))
            return {errors}
	}
    return {};
}
```

But the error message is displayed to the user.

### What version of Directus are you using?

9.22.1

### What version of Node.js are you using?

16.16.0

### What database are you using?

SQLite 3.40.0

### What browser are you using?

Safari

### How are you deploying Directus?

running locally

**Comment by br41nslug** [comment]
This should have either been an `action` or should ahve have updated its output, nice catch 😄

**Comment by llindemann** [comment]
From what I gather, this is due to a small bug in the error-handler, corrected by #17003. It works for my use cases, but maybe I miss some side effects
